xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/objects/classes/BaseClass.java

```diff
@@ -898,6 +909,11 @@ public boolean addStaticListField(String fieldName, String fieldPrettyName, int
             if (defaultValue != null) {
                 list_class.setDefaultValue(defaultValue);
             }
+            list_class.setFreeText(freeText);
+            if (!freeText.equals(ListClass.FREE_TEXT_FORBIDDEN)) {
```

Comment:
It's safer to put the *constant* on the left when checking equality, to prevent a NPE.

Reply:
indeed, fixing.